#78 - Final implementation: render mode options + shrunk banner

vpstackhub opened the pull request:
- Implemented working Render Mode wiring (Per Sample / Per Operation / Per Interval)
- Chart now correctly updates with mode changes
- Added compact label showing current render mode, positioned top-left above chart
- This is just a proposal for the label position ;
<img width="1355" height="952" alt="Screenshot 2025-09-22 105659" src="https://github.com/user-attachments/assets/997d6b87-efa7-4956-b48a-e5e0680369d6" />

<img width="601" height="450" alt="Screenshot 2025-09-22 112531" src="https://github.com/user-attachments/assets/629457b7-c406-410c-988f-86c30b0dedb6" />


I'd love James' feedback if this placement works, and I'm open to moving it elsewhere if preferred.

jamesmarkchan commented:
one more thing is we'll want the target branch being merged into to be `dev` just to avoid accidentally merging directly into `main`. :)

vpstackhub commented:
Thanks James good catch I’ll update the PR to target dev instead of main to keep things safe.On Sep 24, 2025, at 10:36 PM, James Mark Chan ***@***.***> wrote:﻿jamesmarkchan left a comment (JDiskMark/jdm-java#91)
one more thing is we'll want the target branch being merged into to be dev just to avoid accidentally merging directly into main. :)

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>

jamesmarkchan commented:
even though you're not planning to merge into dev yet don't forget to change the target branch you'll be merging into to `dev` so we don't accidentally merge into `main`.

<img width="583" height="31" alt="image" src="https://github.com/user-attachments/assets/5febf03b-7068-4040-87df-bfe51a2eccea" />

vpstackhub commented:
Finalized implementation for Issue #78:

Added RenderFrequencyModeAttributeConverter for persistence

Persisted renderMode in BenchmarkOperation

Displayed render mode column in Benchmark Panel

Fixed numBlocks assignment bug in WRITE loop

Verified mode is shown in UI and persisted correctly

Hopefully indentation and form files look good this time

jamesmarkchan commented:
do you want to close this pull request and keep the other one open since this one is targeting to merge into main and the other targets dev? we'll probably want to stabalize the gui before merging though. currently the branch i pulled is not loading historic benchmarks on startup and there seems to be more spacing between the controls

<img width="216" height="288" alt="image" src="https://github.com/user-attachments/assets/058e3314-14a6-4a59-bf0a-82e2390b2faf" />

when compared to the original, here is the spacing in 1.6.2, good to keep the spacing consistent if possible.

<img width="445" height="445" alt="image" src="https://github.com/user-attachments/assets/22700624-232c-4772-9c1b-b444b8756d4d" />
